include/knowhere/comp/thread_pool.h

```diff
@@ -36,7 +40,7 @@ class ThreadPool {
         std::thread
         newThread(folly::Func&& func) override {
             return folly::NamedThreadFactory::newThread([&, func = std::move(func)]() mutable {
-                if (setpriority(PRIO_PROCESS, gettid(), 19) != 0) {
+                if (setpriority(PRIO_PROCESS, static_cast<pid_t>(syscall(SYS_gettid)), 19) != 0) {
```

Comment:
I'd rather go with the following solution
```
#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 30
#include <sys/syscall.h>
#define gettid() syscall(SYS_gettid)
#endif
```
as suggested https://stackoverflow.com/questions/30680550/c-gettid-was-not-declared-in-this-scope/